optuna/importance/__init__.py

```diff
@@ -78,13 +81,45 @@ def get_param_importances(
 
     Returns:
         An :class:`collections.OrderedDict` where the keys are parameter names and the values are
-        assessed importances.
+        assessed importances. The importances will be sorted in a descending order.
 
     """
+
+    if len(study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))) == 0:
+        raise ValueError("The study does not contain completed trials.")
```

Comment:
I guess this check actually causes performance degradation of optuna-fast-fanova. Can we avoid to issue an additional SQL query here?

Reply:
In my understanding, if we use `CachedStorage`, the results of `get_trials` are cached so that we won't need to fetch all data every time.

Reply:
Yes, it's true if we use CachedStorage.

Reply:
We could change the first argument of `BaseImportanceEvaluator.evaluate` from `study` to `trials: List[FrozenTrial]`, and this would remove the duplicate of both SQL commands and the code. However, I think it is good to do that in a followup PR, since that is going to be another large change. What do you think? @c-bata

Reply:
#3664 Here is the followup PR :)